distribution/pull_v2.go

```diff
@@ -858,7 +859,10 @@ func (p *v2Puller) pullManifestList(ctx context.Context, ref reference.Named, mf
 
 func (p *v2Puller) pullSchema2Config(ctx context.Context, dgst digest.Digest) (configJSON []byte, err error) {
 	blobs := p.repo.Blobs(ctx)
-	configJSON, err = blobs.Get(ctx, dgst)
+	err = retry(ctx, 5, 250*time.Millisecond, func(ctx context.Context) (err error) {
```

Comment:
As to hard-coding max attempts and interval; hard-coding the value is probably ok for a start, but I'd suggest to at least define `consts` to make it slightly more descriptive; the `xfer` package uses a  `maxDownloadAttempts` const (not sure if we want to export that one, and the name is a _bit_ poorly chosen (should probably have been named `defaultXXX`), but perhaps ok (for now) to use the same name.

https://github.com/moby/moby/blob/65b8bcc3216ec64c8f761f35c6caba109b788254/distribution/xfer/download.go#L19

As a follow-up (?) we could look if it's somehow possible to pass the same configuration as `xfer` is using (could be set as a property on `v2Puller` perhaps, but I'd have to have another look); see `xfer.WithMaxDownloadAttempts(config.MaxDownloadAttempts)`; https://github.com/moby/moby/blob/65b8bcc3216ec64c8f761f35c6caba109b788254/daemon/images/service.go#L62

Reply:
I've pulled out the constants, as a first pass here